components/arm/wrapper/wrapper.go

```diff
@@ -85,6 +88,24 @@ func NewWrapperArm(cfg config.Component, r robot.Robot, logger golog.Logger) (ar
 	}, nil
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (wrapper *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		// there is case where ok == true but newCfg.ModelFilePath == ""
+		// because newCfg.ArmName is required as well.
+		if newCfg.ModelFilePath != "" {
+			if model, err := referenceframe.ModelFromPath(newCfg.ModelFilePath, ""); err != nil {
+				wrapper.logger.Debugf("invalid model file path:", err.Error())
```

Comment:
i'd add a comment that it's highly unlikely for this error to happen given it's checked in validate